### PR TITLE
[DC-1629|DC-1633] - Fix error messages output as per the story descriptions

### PR DESCRIPTION
**Pull request by elabeca**
This will apply a fix where the message service raised errors (including getMetadata) didn't return the expected output.

### CODE REVIEW

File: app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala [hunk 1]
```diff
@@ -25,29 +25,45 @@ import play.api.http.Status._
...
       metadata <- messageConnector.getMessageMetadata(replyTo)
       body = createJsonForReply(randomUUID.toString, metadata, twoWayMessageReply, replyTo)
-      response <- messageConnector.postMessage(body)
-    } yield handleResponse(response)
+      resp <- messageConnector.postMessage(body)
+    } yield resp
+
+    response map {
+      handleResponse
+    } recover {
```
Comment (elabeca) [inline app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala]: Will abstract this bit of exception handling in the following PR for DC-1632

File: app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala [hunk 2]
```diff
@@ -25,29 +25,45 @@ import play.api.http.Status._
...
   implicit val hc = HeaderCarrier()
 
   def post(twoWayMessage: TwoWayMessage): Future[Result] = {
     val body = createJsonForMessage(randomUUID.toString, twoWayMessage)
-    messageConnector.postMessage(body) map handleResponse
+    val response = messageConnector.postMessage(body)
+    response map {
+      handleResponse
+    } recover {
+      case e: Upstream4xxResponse => BadGateway(Json.obj("error" -> e.upstreamResponseCode, "message" -> e.message))
```
Comment (elabeca) [inline app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala]: Will abstract this bit of exception handling in the following PR for DC-1632

File: app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala (hunk 1 above)
Comment (elabeca) [reply app/uk/gov/hmrc/twowaymessage/services/TwoWayMessageService.scala]: Will also add respective unit tests in DC-1632. This PR is to unblock Vijay with testing DC-1627.

File: app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala [hunk 3]
```diff
@@ -42,7 +42,7 @@ class TwoWayMessageController @Inject()(twms: TwoWayMessageService)
   def validateAndPostMessage(requestBody: JsValue): Future[Result] =
     requestBody.validate[TwoWayMessage] match {
       case _: JsSuccess[_] => twms.post(requestBody.as[TwoWayMessage])
-      case e: JsError => Future.successful(BadRequest(Json.obj("error" -> "OK", "message" -> JsError.toJson(e))))
+      case e: JsError => Future.successful(BadRequest(Json.obj("error" -> "400", "message" -> JsError.toJson(e))))
```
Comment (sammhmrc) [inline app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala]: The response code should come from the type of response (I.e the BadRequest object here). I think the error should be more descriptive. 

I.e.: "Invalid message" - with perhaps a bit of info.

File: app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala [hunk 4]
```diff
@@ -54,6 +54,6 @@ class TwoWayMessageController @Inject()(twms: TwoWayMessageService)
   def validateAndPostAdvisorResponse(requestBody: JsValue, replyTo: String): Future[Result] =
     requestBody.validate[TwoWayMessageReply] match {
       case _: JsSuccess[_] => twms.postReply(requestBody.as[TwoWayMessageReply], replyTo)
-      case e: JsError => Future.successful(BadRequest(Json.obj("error" -> "OK", "message" -> JsError.toJson(e))))
+      case e: JsError => Future.successful(BadRequest(Json.obj("error" -> "400", "message" -> JsError.toJson(e))))
```
Comment (sammhmrc) [inline app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala]: Same as above.

File: app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala (hunk 3 above)
Comment (elabeca) [reply app/uk/gov/hmrc/twowaymessage/controllers/TwoWayMessageController.scala]: I agree but only going by what the story states. So unable to change in this case.

**Pull request merged**